packages/compass-home/src/components/home.tsx

```diff
@@ -315,6 +331,75 @@ function Home({
     [setIsWelcomeOpen, appRegistry]
   );
 
+  useEffect(() => {
+    function onAutoupdateStarted() {
+      openToast('update-download', {
+        variant: 'progress',
+        title: 'Compass update is in progress',
+      });
+    }
+    function onAutoupdateFailed() {
+      closeToast('update-download');
+      openToast('update-download', {
+        variant: 'warning',
+        title: 'Failed to download Compass update',
+        description: 'Downloading a newer Compass version failed',
+      });
+    }
+    function onAutoupdateSucess() {
+      openToast('update-download', {
+        variant: 'note',
+        title: 'Restart to start newer Compass version',
+        description: (
+          <div className={restartPromptToastStyles}>
+            <div>
+              Continuing to use Compass without restarting may cause some of the
+              features to not work as intended.
+            </div>
+            <div>
+              <Button
+                variant={ButtonVariant.Primary}
+                onClick={() => {
+                  void ipcRenderer?.call(
+                    'autoupdate:update-download-restart-confirmed'
+                  );
+                }}
+              >
+                Restart Compass
+              </Button>
+            </div>
+          </div>
+        ),
+        dismissible: true,
+        onClose: () => {
+          void ipcRenderer?.call(
+            'autoupdate:update-download-restart-dismissed'
+          );
+        },
+      });
+    }
+    ipcRenderer?.on(
+      'autoupdate:update-download-in-progress',
+      onAutoupdateStarted
+    );
+    ipcRenderer?.on('autoupdate:update-download-failed', onAutoupdateFailed);
+    ipcRenderer?.on('autoupdate:update-download-success', onAutoupdateSucess);
+    return () => {
+      ipcRenderer?.removeListener(
+        'autoupdate:update-download-in-progress',
+        onAutoupdateStarted
+      );
+      ipcRenderer?.removeListener(
+        'autoupdate:update-download-failed',
+        onAutoupdateFailed
+      );
+      ipcRenderer?.removeListener(
+        'autoupdate:update-download-success',
+        onAutoupdateSucess
+      );
+    };
+  }, []);
```

Comment:
I pinged you in tech design, seems like we haven't expanded there, but in the initial draft I added a sentence that it would be nice to abstract this behind an interface, yeah. I'm not sure about making toast specifically part of the interface, but some way of getting answers back from renderer  in main that does all the wiring of events on its own would be nice I think

Reply:
Yeah ... fair point, I was thinking about toasts specifically but even just better ways of doing these "reverse" calls would be nice.

I opened https://jira.mongodb.org/browse/COMPASS-7674, let's maybe first figure out what our solution should look like here before we implement it